docs/contract.yaml

```diff
@@ -0,0 +1,156 @@
+      description: |
+      perform a union on two specs, with optional defaults file and conflict resolutions.
+      The endpoint may return a success, which is the result of a successful union,
+      or a failure, which means some unhandled conflicts occurred in the union. In the latter case,
+      a 406 code will be sent along with an array of conflict objects. The user now has two ways
+      to resolve the conflicts. First, they may iterate through the conflicts and add a value to the
+      resolvedValue property based on the options available. They could also manually resolve the
+      conflicts in the original YAML files. In either case, they can send another request to this
+      endpoint with the conflicts resolved.
+      operationId: specMathUnion
+      requestBody:
+        description: data needed for the union operation
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/MergeBody"
+      responses: 
+        '200':
+          description: operation was successful, returns the resultant spec 
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/OperationSuccess"
+        '406':
+          description: return information relevant to the conflict in case of a failed union
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  conflicts:
+                    type: array
+                    items:
+                      "$ref": "#/components/schemas/MergeConflictObject"
+        default:
+          description: error
+          content:
+            application/json:
+              schema:
+                "$ref":  "#/components/schemas/Error"  
+
+  /specMath/overlay:
+    post:
+      description: |
+        the overlay endpoint takes a spec and an overlay file (which is a partial spec)
+        and performs a specialized union where the overlay takes priority. In case of a collision, a
+        conflict will not be reported and the value from the overlay will be taken.
+      operationId: specMathOverlay
+      requestBody:
+        description: data needed for the overlay operation
+        required: true
+        content: 
+          application/json:
+            schema: 
+              "$ref": "#/components/schemas/OverlayBody"
+      responses:
+        '200':
+          description: operation was successful, returns the resultant spec 
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/OperationSuccess"   
+        default:
+          description: error
+          content:
+            application/json:
+              schema:
+                "$ref":  "#/components/schemas/Error"                
+components:
+  schemas:
+    MergeBody:
+      description: holds data required for the union operation
```

Comment:
The meaning of "Body" is unclear to me. There's not really an entity (in my mind) that would own this as its body. "MergeOperands" seems better, but since this seems to exist to represent a request to the merge operation, "MergeRequest" seems best - and also is most consistent with [Google's style guidance](https://google.aip.dev/131).